File: data/code_pairs/cpp/2096.cpp
#include <fstream>
#include <iomanip>

void writesta(int istep, int j, int icutb, int l, double ttime, double time, double dtime) {
    std::ofstream fileStream("unit8.txt", std::ios_base::app);

    if (!fileStream.is_open()) {
        std::cerr << "Failed to open file." << std::endl;
        return;
    }

    fileStream << std::fixed << std::setprecision(6)
               << " " << std::setw(5) << istep
               << " " << std::setw(10) << j
               << " " << std::setw(5) << icutb + 1
               << " " << std::setw(5) << l
               << " " << std::setw(13) << std::scientific << ttime + time
               << " " << std::setw(13) << std::scientific << time
               << " " << std::setw(13) << std::scientific << dtime << std::endl;

    fileStream.close();
}

File: data/code_pairs/fortran/2096.f90
subroutine writesta(istep, j, icutb, l, ttime, time, dtime)
    implicit none

    integer istep, j, icutb, l
    real*8 ttime, time, dtime

    write(8,100) istep, j, icutb+1, l, ttime+time, time, dtime
    flush(8)

100 format(1x,i5,1x,i10,2(1x,i5),3(1x,e13.6))

end subroutine writesta